nuxeo-services/nuxeo-platform-web-common/src/main/resources/web/nuxeo.war/login.jsp

```diff
@@ -526,35 +540,40 @@ a.mobileAppLink:hover {
     <fmt:message bundle="${messages}" key="label.mobile.openInApp" />
   </a>
 </div>
-<% } %>
 
 <script type="text/javascript">
-  <% if (displayMobileBanner) { %>
   // Build mobile app links if the mobile banner is displayed
-  var indexOfHash = window.location.href.indexOf('#!');
-  if (indexOfHash > -1) {
-    // lastPart = #!/doc/default/f6fa9686-3618-47a8-9419-ff1cc76fc857
-    // or lastPart = #!/doc/f6fa9686-3618-47a8-9419-ff1cc76fc857
-    var lastPart = window.location.href.substring(indexOfHash);
-
-    var docPart;
-    var parts = lastPart.split('/');
+  <% if (StringUtils.isBlank(requestedUrlFragment)) { %>
+
+  var urlFragment = decodeURIComponent('<%=requestedUrlFragment%>');
+  var docPart;
+  if (urlFragment.startsWith('!/browse')) {
+    // no repository name for 'browse' URL
+    docPart = urlFragment.replace('!/browse/', '<%=defaultRepositoryName%>/path/');
+  } else {
+    // !/doc/ URL
+    var parts = urlFragment.split('/');
     if (parts.length === 3) {
       // no server in URL
-      docPart = "default/id/" + parts[2];
+      docPart = "<%=defaultRepositoryName%>/id/" + parts[2];
     } else if (parts.length === 4) {
       docPart = parts[2] + "/id/" + parts[3];
     }
-    if(docPart) {
-      var androidAppLink = document.getElementById('androidAppLink');
-      var iOSAppLink = document.getElementById('iOSAppLink');
-      androidAppLink.href += docPart;
-      iOSAppLink.setAttribute('data-action', iOSAppLink.getAttribute('data-action') + docPart);
-    }
   }
+
+  if (docPart) {
+    var androidAppLink = document.getElementById('androidAppLink');
+    var iOSAppLink = document.getElementById('iOSAppLink');
+    androidAppLink.href += docPart;
```

Comment:
Why are we using `var` if we're not doing an undefined or null check ?

Reply:
To avoid polluting the global namespace?

Reply:
Well, the `androidAppLink` element is always here, as it's in the `login.jsp` page itself so the null check is not really needed.
Setting those in a `var` eases the comprehension (imo) instead of chaining stuff to `document.getElementById`... and anyway, it's old code working - not the subject of this PR :)